PT-2418: Fix column data loss during RENAME COLUMN operations

marek-knappe opened the pull request:
The pt-online-schema-change tool was not properly handling RENAME COLUMN syntax (MySQL 8.0+), causing renamed columns to be excluded from the data copy operation. This resulted in NULL values for renamed columns after the schema change.

The issue was in the find_renamed_cols() function which only supported CHANGE COLUMN syntax but not RENAME COLUMN syntax. Added support for RENAME COLUMN parsing to properly detect column renames and include them in the common_cols list for data copying.

Fixes: PT-2418

- [X] The contributed code is licensed under GPL v2.0
- [X] Contributor Licence Agreement (CLA) is signed
- [X] util/update-modules has been ran
- [X] Documentation updated
- [X] Test suite update

marek-knappe committed:
PT-2418: Fix column data loss during RENAME COLUMN operations

The pt-online-schema-change tool was not properly handling RENAME COLUMN
syntax (MySQL 8.0+), causing renamed columns to be excluded from the
data copy operation. This resulted in NULL values for renamed columns
after the schema change.

The issue was in the find_renamed_cols() function which only supported
CHANGE COLUMN syntax but not RENAME COLUMN syntax. Added support for
RENAME COLUMN parsing to properly detect column renames and include
them in the common_cols list for data copying.

Fixes: PT-2418

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/signed)](https://cla-assistant.percona.com:443/percona/percona-toolkit?pullRequest=976) <br/>All committers have signed the CLA.

svetasmirnova committed:
PT-2418 - pt-online-schema-change 3.7.0 lost data when exe alter xxx rename column xxx

- Added clean up to the test
- Added test for enabled option --check-alter
- Removed extra spaces

svetasmirnova committed:
PT-2418 - pt-online-schema-change 3.7.0 lost data when exe alter xxx rename column xxx

- Added instructions to skip the test if server version is less than 8.0

svetasmirnova commented:
Thank you for the contribution. I added cleanup for the test and additional checks. Please let us know if you disagree.

EvgeniyPatlan commented on bin/pt-online-schema-change:
```diff
@@ -11237,6 +11239,17 @@ sub find_renamed_cols {
+      my (undef, $orig_tbl) = Quoter->split_unquote($orig);
+      my (undef, $new_tbl)  = Quoter->split_unquote($new);
+      # Silly but plausible: RENAME COLUMN same_name TO same_name ...
+      next if lc($orig_tbl) eq lc($new_tbl);
+      $renames{lc($orig_tbl)} = $new_tbl;
```
I can only suggest an improvement here. We are doing lower case several times. 
It might be possible to introduce new variables and work with them. Here is the example of the change:

```
my $orig_tbl_lc = lc($orig_tbl);
my $new_tbl_lc  = lc($new_tbl);
next if $orig_tbl_lc eq $new_tbl_lc;
$renames{$orig_tbl_lc} = $new_tbl;
```

marek-knappe replied:
I just followed the same convention used in the block above.
I don’t think introducing new variables is necessary here — it doesn’t save any real CPU and adds extra lines that make the code slightly harder to read, but if that's the wish of owners I'm happy to change it :)
